aten/src/ATen/gen.py

```diff
@@ -217,13 +241,33 @@ def format_yaml(data):
     return yaml.dump(data, default_flow_style=False, Dumper=noalias_dumper, width=float('Inf'))
 
 
-def generate_storage_type_and_tensor(backend, density, declarations):
+def add_op_registrations(per_type_registrations, per_op_registrations, op_registrations):
+    for op_registration in op_registrations:
+        opname = op_registration.operator_name
+        registration = op_registration.registration_code
+        # apply whitelist
+        if op_registration_whitelist and opname not in op_registration_whitelist:
+            continue
+        # per type registration
+        if not options.per_op_registration:
+            per_type_registrations.append(registration)
+            continue
+        # per op registration
+        try:
+            per_op_registrations[opname].append(registration)
```

Comment:
Seems like a good use case for defaultdict :>

Reply:
updated